### PR TITLE
Added LoggingExporter for AppSignals during translation time if debugging is enabled.

### PR DESCRIPTION
**Pull request by musa-asad**
Note: This merges https://github.com/aws/amazon-cloudwatch-agent/pull/1148 into `main`.

# Description of the issue
There is no LoggingExporter for AppSignals during translation time.

When debugging is enabled, we want to add the LoggingExporter with detailed verbosity to the translated `yaml`.

# Description of changes
During translation:
- Adds LoggingExporter as an exporter.
- Adds LoggingExporter to AppSignals pipelines when debugging is enabled.

# License
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

# Tests
- Ran unit tests: `common_test.go`, `logging/translator_test.go`, and `appsignals/translator_test.go`.
  - `common_test.go` was just run to ensure that unit test still works for the new changes in `common.go`.
  - `logging/translator_test.go` checks to see if the `logging.go` is set up correctly and sees if the agent key missing throws an error and if enabling debugging provides the correct yaml configuration for the loggingexporter.
  - `appsignals/translator_test.go` includes a new check to see if the loggingexporter is appropriately added to the exporters when both appsignals and debugging are enabled.

# Requirements
_Before commit the code, please do the following steps._
1. Run `make fmt` and `make fmt-sh`
2. Run `make lint`

### CODE REVIEW

File: translator/translate/otel/common/common.go [hunk 1]
```diff
@@ -80,6 +81,8 @@ var (
 		component.DataTypeTraces:  {AppSignalsTraces, AppSignalsTracesFallback},
 		component.DataTypeMetrics: {AppSignalsMetrics, AppSignalsMetricsFallback},
 	}
+
+	DebugLogging = ConfigKey(AgentKey, DebugKey)
```
Comment (jefchien) [inline translator/translate/otel/common/common.go:85]:
```suggestion
	AgentDebugConfigKey = ConfigKey(AgentKey, DebugKey)
```

File: translator/translate/otel/exporter/logging/translator_test.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	testCases := map[string]struct {
+		input   map[string]interface{}
+		want    *confmap.Conf
+		wantErr error
+	}{
+		"WithMissingKey": {
+			input: map[string]interface{}{"logs": map[string]interface{}{}},
+			wantErr: &common.MissingKeyError{
+				ID:      tt.ID(),
+				JsonKey: common.ConfigKey(common.AgentKey, common.DebugKey),
```
Comment (jefchien) [inline translator/translate/otel/exporter/logging/translator_test.go:39]: nit: 
```suggestion
				JsonKey: common.DebugLogging,
```

File: translator/translate/otel/exporter/logging/translator_test.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+			},
+		},
+		"WithDebugLoggingEnabled": {
+			input: map[string]interface{}{
+				"agent": map[string]interface{}{
+					"debug": true,
+				},
+			},
+			want: testutil.GetConf(t, filepath.Join("testdata", "config.yaml")),
```
Comment (jefchien) [inline translator/translate/otel/exporter/logging/translator_test.go:48]: nit: Don't need a YAML file for this.

```suggestion
			want: confmap.NewFromStringMap(map[string]any{
			   "verbosity": "detailed"
			})
```

File: translator/translate/otel/pipeline/applicationsignals/translator_test.go [hunk 4]
```diff
@@ -132,6 +132,28 @@ func TestTranslatorMetricsForKubernetes(t *testing.T) {
...
 			isEKSCache: eksdetector.TestIsEKSCacheEKS,
 		},
+		"WithAppSignalsAndLoggingEnabled": {
+			input: map[string]interface{}{
+				"agent": map[string]interface{}{
+					"debug": true,
+				},
+				"logs": map[string]interface{}{
+					"metrics_collected": map[string]interface{}{
+						"app_signals": map[string]interface{}{
```
Comment (lisguo) [inline translator/translate/otel/pipeline/applicationsignals/translator_test.go:142]: we should use `application_signals` as the new official config

**Review by jefchien — DISMISSED** [review]
Make sure to squash your commits and clear the description when you merge.

File: translator/translate/otel/pipeline/applicationsignals/translator_test.go [hunk 5]
```diff
@@ -207,6 +229,28 @@ func TestTranslatorMetricsForEC2(t *testing.T) {
 			detector:   eksdetector.TestEKSDetector,
 			isEKSCache: eksdetector.TestIsEKSCacheEKS,
 		},
+		"WithAppSignalsAndLoggingEnabled": {
```
Comment (lisguo) [inline translator/translate/otel/pipeline/applicationsignals/translator_test.go:232]: I think we are missing translation tests that actually test against yaml. Maybe we can extend this config? https://github.com/aws/amazon-cloudwatch-agent/blob/main/translator/tocwconfig/sampleConfig/base_appsignals_config.yaml

File: translator/translate/otel/exporter/logging/translator_test.go [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+				JsonKey: common.AgentDebugConfigKey,
+			},
+		},
+		"WithDebugLoggingEnabled": {
+			input: map[string]interface{}{
+				"agent": map[string]interface{}{
+					"debug": true,
+				},
+			},
+			want: confmap.NewFromStringMap(map[string]any{
```
Comment (jefchien) [inline translator/translate/otel/exporter/logging/translator_test.go:46]: nit: You also don't need to unmarshal this is you can just create a config.

```suggestion
			want: &loggingexporter.Config{Verbosity: configtelemetry.LevelDetailed}
```

File: translator/translate/otel/pipeline/applicationsignals/translator_test.go (hunk 4 above)
Comment (jefchien) [reply translator/translate/otel/pipeline/applicationsignals/translator_test.go:142]: Not sure how this passed unit testing without that change.

File: translator/translate/otel/pipeline/applicationsignals/translator.go [hunk 7]
```diff
@@ -54,6 +55,10 @@ func (t *translator) Translate(conf *confmap.Conf) (*common.ComponentTranslators
 	translators.Processors.Set(resourcedetection.NewTranslator(resourcedetection.WithDataType(t.dataType)))
 	translators.Processors.Set(awsapplicationsignals.NewTranslator(awsapplicationsignals.WithDataType(t.dataType)))
 
+	if enabled, _ := common.GetBool(conf, common.DebugLogging); enabled {
```
Comment (jefchien) [inline translator/translate/otel/pipeline/applicationsignals/translator.go:58]: Need to update the name everywhere.

File: translator/translate/otel/pipeline/applicationsignals/translator_test.go (hunk 4 above)
Comment (lisguo) [reply translator/translate/otel/pipeline/applicationsignals/translator_test.go]: we allow for both app_signals and application_signals, but after speaking with the pulse team they are okay with deprecating app_signals but we haven't made that change

File: translator/translate/otel/pipeline/applicationsignals/translator_test.go [hunk 8]
```diff
@@ -132,11 +132,33 @@ func TestTranslatorMetricsForKubernetes(t *testing.T) {
...
+			input: map[string]interface{}{
+				"agent": map[string]interface{}{
+					"debug": true,
+				},
+				"logs": map[string]interface{}{
+					"metrics_collected": map[string]interface{}{
+						"application_signals": map[string]interface{}{
+							"enabled": true,
```
Comment (lisguo) [inline translator/translate/otel/pipeline/applicationsignals/translator_test.go:143]: nit: Don't need this line

File: translator/translate/otel/pipeline/applicationsignals/translator_test.go [hunk 9]
```diff
@@ -207,6 +229,28 @@ func TestTranslatorMetricsForEC2(t *testing.T) {
...
```
Comment (lisguo) [inline translator/translate/otel/pipeline/applicationsignals/translator_test.go:240]: nit: Remove line

**Review by lisguo — DISMISSED** [review]

File: translator/tocwconfig/tocwconfig_test.go [hunk 10]
```diff
@@ -76,7 +76,7 @@ func TestGenericAppSignalsConfig(t *testing.T) {
 	context.CurrentContext().SetMode(config.ModeOnPremise)
 	t.Setenv(config.HOST_NAME, "host_name_from_env")
 	t.Setenv(config.HOST_IP, "127.0.0.1")
-	expectedEnvVars := map[string]string{}
+	expectedEnvVars := map[string]string{"CWAGENT_LOG_LEVEL": "DEBUG"}
```
Comment (jefchien) [inline translator/tocwconfig/tocwconfig_test.go:79]: This shouldn't be necessary. You're explicitly setting the field to true in the JSON already.

**Review by jefchien — APPROVED** [review]

File: translator/tocwconfig/tocwconfig_test.go (hunk 10 above)
Comment (jefchien) [reply translator/tocwconfig/tocwconfig_test.go:79]: It looks like we're setting the environment variables as part of `ToEnvConfig`

**Review by lisguo — APPROVED** [review]

**Pull request merged**